pkgs/pkgs-lib/formats.nix

```
@@ -142,6 +142,63 @@ rec {
 
   };
 
+  iniWithGlobalSection = {
+    # Represents lists as duplicate keys
+    listsAsDuplicateKeys ? false,
+    # Alternative to listsAsDuplicateKeys, converts list to non-list
+    # listToValue :: [IniAtom] -> IniAtom
+    listToValue ? null,
+    ...
+    }@args:
+    assert !listsAsDuplicateKeys || listToValue == null;
```

IMO slightly easier to read with an implication:

```suggestion
    assert listsAsDuplicateKeys -> listToValue == null;
```

---

Agreed. I just copy&pasted that part of the code, so I didn't give it any second thought. I changed it throughout the file now.